src/Plugin/Commerce/PaymentGateway/Heartland.php

```diff
@@ -336,14 +353,43 @@ public function refundPayment(PaymentInterface $payment, Price $amount = null)
      */
     public function createPaymentMethod(PaymentMethodInterface $payment_method, array $payment_details)
     {
-        // $payment_details['token_value'] = NULL;
         if (empty($payment_details['token_value'])) {
             throw new \InvalidArgumentException(t('token_value is not set in $payment_details.'));
         }
 
-        // Keeps card data from being stored during processing.
-        // Will need to update this for iteration 2, multi-use tokenization.
-        $payment_method->setReusable(false);
+        // Retrieve a multi-use token and store non-sensitive card info if subscriptions option is enabled
+        // *** Multi-use tokenization must be enabled on your Heartland merchant account for this to work ***
+        if ($this->configuration['subscriptions']) {
+
+            // Enable Drupal to store non-sensitive card data
+            $payment_method->setReusable(true);
+
+            $this->authenticate();
+
+            // Heartland SDK - Prepare / Charge Credit Card
+            // https://developer.heartlandpaymentsystems.com/Documentation/credit-card-payments/#prepare-to-charge-a-credit-card
+            $card = new CreditCardData();
+            $card->token = $payment_details['token_value'];
+
+            try {
+                // Heartland SDK - Requesting a multi-use token
+                // https://developer.heartlandpaymentsystems.com/Documentation/credit-card-payments/#using-multi-use-tokens
+                $response = $card->tokenize()
```

Comment:
Are we able to pass the billing address during this call? This will fail for American Express card holders as Amex requires at least a billing zip code to be passed as they do not support a true account verify.